### PR TITLE
Moray/readme metamask + programmatic deployment 

### PR DESCRIPTION
**Pull request by moraygrieve**
### Why is this change needed?

- Adds a section to the top level readme that references docs on how to deploy a contract with metamask
- Adds a section to demonstrate how to programmatically deploy and interact with a contract on to Obscuro Testnet

### What changes were made as part of this PR:

- update the top level readme for referencing docs on using metamask for manual deployment 
- added deploying-a-smart-contract-programmatically.md and deploying-a-smart-contract-programmatically.py into docs/testnet and referenced the md file in the top level readme. 

### What are the key areas to look at

- check that deploying-a-smart-contract-programmatically.md supplies enough information to be useful to a user and whether more detail is required

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -361,7 +361,34 @@ account on the L1 network used to deploy the Obscuro Management and the ERC20 co
...
+Deploying and interacting with contracts on Obscuro requires the wallet extension to be running. The wallet extension is the Obscuro 
+component that ensures that sensitive information in RPC requests between client applications and Obscuro cannot be seen by third parties. 
+The wallet extension should be run local to the client application and is described in more detail at 
+[docs/testnet/wallet-extension.md](docs/testnet/wallet-extension.md).
+
+To start the wallet extension to run against a local testnet use the below;
+
+```
+cd ./tools/walletextension/main/
+go build -o wallet_extension
```
Comment (otherview) [inline README.md:376]: Small nit. Might be wrong but I thought it required a directory. Like `.` turning into `go build -o wallet_extension .`

**Review by otherview — APPROVED** [review]

**Review by BedrockSquirrel — COMMENTED** [review]
Instructions and example LGTM, but did you mean to commit that wallet_extension binary?

File: README.md (hunk 1 above)
Comment (moraygrieve) [reply README.md:376]: I think it defaults to working directory if not give ... I don't need to give one when I run the command but can change to make it more explicit if needed?

Comment (otherview) [reply README.md:376]: No need, I didn't know there was that default :)

**Pull request merged**